### PR TITLE
Dast Automation General Settings Changes

### PR DESCRIPTION
**Pull request by SmitGala**

### CODE REVIEW

File: app/components/project-settings/general-settings/device-preferences-automated-dast/index.scss [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      --project-settings-general-settings-device-preferences-automated-dast-border-color
+    );
+  width: 100%;
+  max-width: 600px;
+  gap: 0.57em;
+  padding: 0.57em 0;
+  margin-top: 1.285em;
+}
+
+.version-select {
```
Comment (Yibaebi) [inline app/components/project-settings/general-settings/device-preferences-automated-dast/index.scss:19]: class not getting used

File: app/components/project-settings/general-settings/device-preferences-automated-dast/index.hbs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            @alignItems='center'
+            data-test-dast-preference-criteria-table
+          >
+            <AkStack @width='4/12' class='ml-2'>
+              <AkTypography>{{t 'minOSVersion'}}</AkTypography>
+            </AkStack>
+
+            {{#if this.loaded}}
+              <AkSelect
+                @placeholder='Select Min OS Version'
```
Comment (Yibaebi) [inline app/components/project-settings/general-settings/device-preferences-automated-dast/index.hbs:86]: Translate text.

File: app/components/project-settings/general-settings/device-preferences-automated-dast/index.hbs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        <AkTypography
+          @variant='h5'
+          data-test-projectSettings-generalSettings-devicePreferenceAutomatedDast-title
+        >
+          {{t 'devicePreferencesAutomatedDast'}}
+        </AkTypography>
+      {{/if}}
+
+      <AkSelect
+        @placeholder={{'Select the device preference'}}
```
Comment (Yibaebi) [inline app/components/project-settings/general-settings/device-preferences-automated-dast/index.hbs:21]: Translate text here also

File: app/components/project-settings/general-settings/device-preferences-automated-dast/index.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    selectedDevice?: ProfileDsAutomatedDeviceSelection
+  ) {
+    return this.deviceSelectionTypes.find((st) => st.value === selectedDevice);
+  }
+
+  @action
+  getDeviceCapabilitiesOptionsData(profile?: ProfileModel | null) {
+    return [
+      {
+        label: 'SIM',
```
Comment (Yibaebi) [inline app/components/project-settings/general-settings/device-preferences-automated-dast/index.ts:47]: Translate

File: tests/integration/components/project-settings/general-settings/device-preferences-automated-dast-test.js [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        return {
+          ds_automated_device_selection:
+            ENUMS.DS_MANUAL_DEVICE_SELECTION.ANY_DEVICE,
+        };
+      });
+
+      this.server.get(
+        '/profiles/:id/ds_automated_device_preference',
+        (schema) => {
+          console.log(schema);
```
Comment (Yibaebi) [inline tests/integration/components/project-settings/general-settings/device-preferences-automated-dast-test.js:63]: Remove log

File: tests/integration/components/project-settings/general-settings/device-preferences-automated-dast-test.js [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+
+module(
+  'Integration | Component | project-settings/general-settings/device-preferences-automated-dast',
+  function (hooks) {
+    setupRenderingTest(hooks);
+    setupMirage(hooks);
+    setupIntl(hooks);
+
+    hooks.beforeEach(async function () {
+      this.server.createList('organization', 1);
```
Comment (Yibaebi) [inline tests/integration/components/project-settings/general-settings/device-preferences-automated-dast-test.js:26]: Arrange `beforeEach` hook properly. Section models, server mocks and property setters appropriately.

File: tests/integration/components/project-settings/general-settings/device-preferences-automated-dast-test.js [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+      selectListItems = findAll('.ember-power-select-option');
+
+      assert.dom(selectListItems[1]).hasAria('selected', 'true');
+
+      await click(`.${classes.trigger}`);
+
+      assert.dom('[data-test-dast-preference-criteria-table]').exists();
+    });
+
+    test('it selects min version', async function (assert) {
```
Comment (Yibaebi) [inline tests/integration/components/project-settings/general-settings/device-preferences-automated-dast-test.js:145]: Test does not check if selected minOSVersion is being sent correctly via the network.

File: tests/integration/components/project-settings/general-settings/device-preferences-automated-dast-test.js [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+        .containsText('t:devicePreferencesAutomatedDast:()');
+
+      assert
+        .dom(
+          '[data-test-projectSettings-generalSettings-devicePreferenceAutomatedDast-preference-select]'
+        )
+        .exists();
+    });
+
+    test('it selects criteria', async function (assert) {
```
Comment (Yibaebi) [inline tests/integration/components/project-settings/general-settings/device-preferences-automated-dast-test.js:108]: Test does not check if selected device criteria and selection is being sent correctly via the network.

File: translations/en.json [hunk 9]
```diff
@@ -758,7 +764,6 @@
       "companyName": "Company Name",
       "appName": "App Name",
       "appEnv": "App Environment",
-      "minOSVersion": "Min OS Version",
```
Comment (future-pirate-king) [inline translations/en.json:761]: does this not impact manual scan modal ?

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/helpers/ds-automated-device-pref.ts [hunk 10]
```diff
@@ -0,0 +1,27 @@
+import { helper } from '@ember/component/helper';
+import ENUMS from 'irene/enums';
+
+export function dsAutomatedDevicePref(params: [number | string]) {
+  const devicePrefernce = params[0];
```
Comment (future-pirate-king) [inline app/helpers/ds-automated-device-pref.ts:5]: typo devicePrefernce

File: app/helpers/ds-manual-device-pref.ts [hunk 11]
```diff
@@ -0,0 +1,27 @@
+import { helper } from '@ember/component/helper';
+import ENUMS from 'irene/enums';
+
+export function dsManualDevicePref(params: [number | string]) {
+  const devicePrefernce = params[0];
```
Comment (future-pirate-king) [inline app/helpers/ds-manual-device-pref.ts:5]: typo devicePrefernce

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/models/profile.ts [hunk 12]
```diff
@@ -1,6 +1,44 @@
...
+export enum ProfileDsAutomatedDeviceSelection {
+  ANY_DEVICE = 0,
+  FILTER_CRITERIA = 1,
+}
+
+export enum ProfileDsAutomatedDeviceType {
+  NO_PREFERENCE = 0,
+  PHONE_REQUIRED = 1,
+  TABLET_REQUIRED = 2,
+}
```
[the remark this answers is not in the document]
Comment (Yibaebi) [reply app/models/profile.ts:24]: For type purposes.

**Pull request merged**